### PR TITLE
Backport to 3.3 Usability test 

### PR DESCRIPTION
**Pull request by marinakog**
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/9848/
build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/9827

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java [hunk 1]
```diff
@@ -843,7 +855,7 @@ void testTwoDomainsInSameNameSpaceOnOperator() {
       assertDoesNotThrow(() ->
               TestActions.scaleClusterWithScalingActionScript(clusterName, domain4Uid, domain4Namespace,
                   "/u01/domains/" + domain4Uid, "scaleDown", 1,
-                  op2Namespace,opServiceAccount),
+                  op3Namespace,opServiceAccount),
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java:858]: add space after comma.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java [hunk 2]
```diff
@@ -852,17 +864,36 @@ void testTwoDomainsInSameNameSpaceOnOperator() {
       assertDoesNotThrow(() ->
               TestActions.scaleClusterWithScalingActionScript(clusterName, domain5Uid, domain4Namespace,
                   "/u01/domains/" + domain5Uid, "scaleDown", 1,
-                  op2Namespace,opServiceAccount),
+                  op3Namespace,opServiceAccount),
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java:867]: add space after comma

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java [hunk 3]
```diff
@@ -852,17 +864,36 @@ void testTwoDomainsInSameNameSpaceOnOperator() {
...
               checkPodDoesNotExist(managedServerPodName2, domain5Uid, domain4Namespace),
           " scaling via scalingAction.sh script was not succeeded for domain5");
       logger.info("Domain5 scaled to 2 servers");
     } finally {
+      try {
+        String operatorPodName =
+                assertDoesNotThrow(() -> getOperatorPodName(OPERATOR_RELEASE_NAME, op3Namespace),
+                        "Can't get operator's pod name");
+        Path logDirPath = Paths.get(RESULTS_ROOT, this.getClass().getSimpleName());
+        assertDoesNotThrow(() -> FileUtils.deleteDirectory(logDirPath.toFile()),
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java:880]: static import FileUtils.deleteDirectory here

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java [hunk 4]
```diff
@@ -852,17 +864,36 @@ void testTwoDomainsInSameNameSpaceOnOperator() {
...
+                "Delete directory failed");
+        assertDoesNotThrow(() -> Files.createDirectories(logDirPath),
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java:882]: static import Files.createDirectories here

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java [hunk 5]
```diff
@@ -852,17 +864,36 @@ void testTwoDomainsInSameNameSpaceOnOperator() {
...
+                "Create directory failed");
+        String podLog = assertDoesNotThrow(() -> TestActions.getPodLog(operatorPodName, op3Namespace));
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java:884]: static import TestActions.getPodLog here

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java [hunk 6]
```diff
@@ -852,17 +864,36 @@ void testTwoDomainsInSameNameSpaceOnOperator() {
...
+        Path pathToLog =
+                Paths.get(RESULTS_ROOT, this.getClass().getSimpleName(),
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java:886]: static import Paths.get here

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java [hunk 7]
```diff
@@ -852,17 +864,36 @@ void testTwoDomainsInSameNameSpaceOnOperator() {
...
+                        "/TwoDomainsInSameNameSpaceOnOperatorOpLog" + op3Namespace + ".log");
+
+        assertDoesNotThrow(() -> Files.write(pathToLog, podLog.getBytes()),
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java:889]: static import Files.write here

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java [hunk 8]
```diff
@@ -852,17 +864,36 @@ void testTwoDomainsInSameNameSpaceOnOperator() {
...
+                "Can't write to file " + pathToLog);
+      } catch (Exception ex) {
+        logger.info("Failed to collect operator log");
+      }
       uninstallOperator(op1HelmParams);
-      deleteSecret(OCIR_SECRET_NAME,op2Namespace);
-      cleanUpSA(op2Namespace);
+      deleteSecret(OCIR_SECRET_NAME,op3Namespace);
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java:895]: add space after comma

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java [hunk 9]
```diff
@@ -986,37 +987,61 @@ private static void scaleViaScript(String opNamespace, String domainNamespace,
...
-            condition -> logger.info("Copying scalingAction.log from admin server pod, waiting for success "
-                    + "(elapsed time {0}ms, remaining time {1}ms)",
-                condition.getElapsedTimeInMS(),
-                condition.getRemainingTimeInMS()))
-        .until(() -> {
-          return copyFileFromPod(domainNamespace, adminPod.getMetadata().getName(), null,
-              domainHomeLocation + "/bin/scripts/scalingAction.log",
-              Paths.get(RESULTS_ROOT + "/" + domainUid + "-scalingAction.log"));
-        });
+    testUntil(
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java:1024]: the following section for copying scalingAction.log and scalingAction.out seems the same as above.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java [hunk 10]
```diff
@@ -986,37 +987,61 @@ private static void scaleViaScript(String opNamespace, String domainNamespace,
...
+            logger,
+            "Copying scalingAction.log from admin server pod");
+    // copy scalingAction.out to local
+    testUntil(
+            () -> copyFileFromPod(domainNamespace, adminPod.getMetadata().getName(), null,
+                    "/u01/scalingAction.out",
+                    Paths.get(RESULTS_ROOT + "/" + domainUid + "-scalingAction.out")),
+            logger,
+            "Copying scalingAction.out from admin server pod");
+    //      domainHomeLocation + "/bin/scripts/scalingAction.log",
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java:1037]: delete commented out lines

**Review by xiancao — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java [hunk 11]
```diff
@@ -829,7 +841,7 @@ void testTwoDomainsInSameNameSpaceOnOperator() {
       logger.info("Domain4 scaled to 3 servers");
 
       assertTrue(scaleClusterWithRestApi(domain5Uid, clusterName,3,
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java:843]: I scale both domain4 and domain5, line 841 is for domain4  after it verified all servers are up

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java (hunk 9 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java:1024]: replaced with private method

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java (hunk 8 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java (hunk 7 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java (hunk 6 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java [hunk 12]
```diff
@@ -829,7 +841,7 @@ void testTwoDomainsInSameNameSpaceOnOperator() {
...
-          externalRestHttpsPort,op2Namespace, opServiceAccount),
+          externalRestHttpsPort,op3Namespace, opServiceAccount),
           "Domain2 " + domain4Namespace + " scaling operation failed");
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java (hunk 1 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java (hunk 3 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java (hunk 2 above)
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java]: done

**Review by maggiehe00 — APPROVED** [review]

**Pull request merged**